### PR TITLE
[SELS-TASK][UI/BE] Signup

### PR DESCRIPTION
**Pull request by rdelrosariojr**
### [Asana Link]
https://app.asana.com/0/1201713837896454/1201725942471527/f
### [Commands]
```
php artisan serve
npm start
```
### [Pre-conditions]
After running npm start and php artisan serve. You should go to the url based on the suggested url on php artisan serve command
### [Expected Output]
Should be able to see the sign in and sign in form
### [Notes]
- `console.log(state.auth);` in Signup.js file is just there to show that signup is working.
- This PR includes field validations.
### [Screenshots]
Screen Shot 2022-02-07 at 3 05 01 PM
Screen Shot 2022-02-07 at 3 07 02 PM

### CODE REVIEW

File: back-end/e-learning/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -31,12 +31,28 @@ public function me()
...
+  }
+
+  public function signup(Request $request)
+  {
+    $user = new User();
+    $user->first_name = $request->first_name;
+    $user->last_name = $request->last_name;
+    $user->email = $request->email;
+    $user->password = Hash::make($request->password);
+    $user->role = 'member';
```
Comment (AubreyV) [inline back-end/e-learning/app/Http/Controllers/UserController.php]: You can use `fill` here to reduce number of times of referencing to `$user`

`fill` is like `update` only that it does not update immediately. This is useful if you still want to have a logic to handle when a user is successfully saved or not. Like what you have on `line 51`

example:
```
// fill() will save the values into memory temporarily until the save() function is called for the user
$user->fill([
  'name' => 'Aubrey Vicente',
  'age' => 25
  'email' => 'aubrey@test.com'
]);

// save() will complete the update process of the user above
$user->save();
```

File: back-end/e-learning/routes/api.php [hunk 2]
```diff
@@ -23,4 +23,5 @@
   Route::post('logout', [UserController::class, 'logout']);
 });
 
-Route::post('login', [UserController::class, 'login']);
+Route::post("login", [UserController::class, 'login']);
+Route::post("signup", [UserController::class, 'signup']);
```
Comment (AubreyV) [inline back-end/e-learning/routes/api.php]: Please make the quote setting standard for all

File: back-end/e-learning/app/Http/Controllers/UserController.php (hunk 1 above)
Comment (rdelrosariojr) [reply back-end/e-learning/app/Http/Controllers/UserController.php]: I will look into this ma'am

**Review by AubreyV — COMMENTED** [review]
## Comment

- [ ] I cannot go to Sign Up Page by clicking the URL in Sign In Page
https://user-images.githubusercontent.com/21002044/154188946-4fae0848-8d60-4e46-9c4a-27ae80ec4072.mov
- [ ] I cannot go to Sign Up Page by clicking the button from the Navbar
https://user-images.githubusercontent.com/21002044/154189289-215542f9-6c39-47fd-8bff-edae904dd40d.mov
 - [ ] I also cannot go to Login Page by clicking Login from the Navbar


Comments: 3
Iterations: 1
Testing Time: 20 mins

**Comment by rdelrosariojr** [comment]
> ## Comment
> * [ ]  I cannot go to Sign Up Page by clicking the URL in Sign In Page
>   https://user-images.githubusercontent.com/21002044/154188946-4fae0848-8d60-4e46-9c4a-27ae80ec4072.mov
> * [ ]  I cannot go to Sign Up Page by clicking the button from the Navbar
>   https://user-images.githubusercontent.com/21002044/154189289-215542f9-6c39-47fd-8bff-edae904dd40d.mov
> * [ ]  I also cannot go to Login Page by clicking Login from the Navbar
> 
> Comments: 3 Iterations: 1 Testing Time: 20 mins

Hi ma'am, i haven't added Routing yet. I forgot to mention it in Notes.
I'll do it nalang karon, I will commit changes for that in Signup PR once i'm done.
Sorry ma'am.

**Comment by rdelrosariojr** [comment]
Or if it is okay ma'am that you may go to http://localhost:3000/signup nalang directly to check Signup UI and Funcationality, then i'll make another PR for Login and Signup routing nlang po.

**Comment by AubreyV** [comment]
> ## Comment
> * [ ]  I cannot go to Sign Up Page by clicking the URL in Sign In Page
>   https://user-images.githubusercontent.com/21002044/154188946-4fae0848-8d60-4e46-9c4a-27ae80ec4072.mov
> * [ ]  I cannot go to Sign Up Page by clicking the button from the Navbar
>   https://user-images.githubusercontent.com/21002044/154189289-215542f9-6c39-47fd-8bff-edae904dd40d.mov
> * [ ]  I also cannot go to Login Page by clicking Login from the Navbar
> 
> Comments: 3 Iterations: 1 Testing Time: 20 mins

This will be fixed in a different Asana Issue: https://app.asana.com/0/1201713837896454/1201834662393276/f

**Review by AubreyV — APPROVED** [review]
## Approve
Sign Up functionality working properly
Can Login using the newly signed up credentials

Comments: 3 (created separate [issue](https://app.asana.com/0/1201713837896454/1201834662393276/f))
Iterations: 1
Testing hours: 30 mins

**Pull request merged**